core/src/actors/blocks_manager/handlers.rs

```diff
@@ -63,3 +67,24 @@ impl Handler<GetHighestBlockCheckpoint> for BlocksManager {
         }
     }
 }
+
+impl Handler<AddNewBlock> for BlocksManager {
+    type Result = ();
+
+    fn handle(&mut self, msg: AddNewBlock, _ctx: &mut Context<Self>) {
+        match self.process_new_block(msg.block) {
+            Ok(hash) => {
+                // Get sessions manager address
+                let sessions_manager_addr = System::current().registry().get::<SessionsManager>();
+
+                let items = vec![witnet_data_structures::chain::InvVector::Block(hash)];
```

Comment:
I would add here a comment to describe what this code does and perhaps add a `use` statement to avoid the full path to the `InvVector`:
```suggestion
                // up in the file header
                use witnet_data_structures::chain::InvVector;
                
                // Send a broadcast message to the SessionsManager to announce the new block
                let items = vec![InvVector::Block(hash)];
```